client/src/components/projects/project-filter/ProjectFilter.vue

```diff
@@ -0,0 +1,178 @@
+                        v-on:update:checked="handleFilterAction(domain, 'domain')"/>
+              <span class="title is-5 is-size-16">{{ domain.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+    <Accordeon
+          v-if="customers && customers.length"
+          name="Customers"
+          :opened="customerAccordionOpened"
+          @update:opened="customerAccordionOpened=!customerAccordionOpened"
+        >
+          <div
+            v-for="customer in convertedCustomers"
+            :key="customer.name"
+          >
+            <div class="filter-item">
+              <Checkbox :checked = "customer.checked"
+                        v-on:update:checked="handleFilterAction(customer, 'customer')" /> 
+              <span class="title is-5 is-size-16">{{ customer.name }}</span>
+            </div>
+          </div>
+    </Accordeon>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import {IProject} from '../../../shared/interfaces/project';
+  import Accordeon from '../../common/Accordeon/Accordeon.vue';
+  import Checkbox from '../../common/Checkbox/Checkbox.vue';
+  import Constants from '../../../shared/constants/constants';
+  import { Extension } from '../../../shared/classes/Extension';
+  
+
+  export default Vue.extend({
+      data() {
+        return {
+            // properties for accordion sections open/closed
+            lineAccordionOpened: true,
+            programAccordionOpened: true,
+            typeAccordionOpened: true,
+            domainAccordionOpened: true,
+            customerAccordionOpened: true,
+            technologyAccordionOpened: true,
+
+            // models for accordeon section
+            lines: new Array(),
+            domains: new Array(),
+            programs: new Array(),
+            types: new Array()
+        };
+      },
+
+      props: ['customers'],
+
+      computed: {
+        // customers come from parent
+        // no customers in constants.ts
+        convertedCustomers(): any[] {
+          return this.createModelForCheckboxes(this.customers);
+        }
+      },
+
+      components: {
+          Accordeon,
+          Checkbox
+      },
+      created() {
+         // convert for accordeon section with checkboxes
+         this.lines = this.createModelForCheckboxes(Constants['lines']);
+         this.domains =  this.createModelForCheckboxes(Constants['domains']);
+         this.types = this.createModelForCheckboxes(Constants['types']);
+         this.programs = this.createModelForCheckboxes(Constants['programs']);
+      },
+      methods: {
+        
+        // rerender checkbox
+        // will be used later for project filtering via store
+        handleFilterAction(item:any, key: string){
+          item.checked = !item.checked;
+          // computed  property convertedCustomers won't render automatically after 
+          // checkbox change
+          if(key === 'customer') this.$forceUpdate();
```

Comment:
There's definately something wrong here. Add TODO to change.
$foceUpdate is rarely needed and item.checked should be changed inside checkbox